### ISSUE TITLE
Constant factor error in JAX 101 docs

### ISSUE THREAD

**Issue by ArthurConmy** [google/jax#18219]
### Description

In the [Advanced Autodiff](https://jax.readthedocs.io/en/latest/jax-101/04-advanced-autodiff.html) section of the JAX 101 docs, there is a bug in the TD(0) section: the **pseudo loss function** is wrong and needs a constant factor of -0.5 throughout this section.

I have implemented a fix for this [here](https://github.com/google/jax/pull/18220) and added an extra code cell to confirm the fix is correct. 

### What jax/jaxlib version are you using?

jaxlib v0.4.19

### Which accelerator(s) are you using?

N/A

### Additional system info

N/A

**Comment by hawkinsp** [comment]
Fixed!
